src/mesh/NodeDB.cpp

```diff
@@ -925,6 +925,8 @@ meshtastic_NodeInfoLite *NodeDB::getOrCreateMeshNode(NodeNum n)
     meshtastic_NodeInfoLite *lite = getMeshNode(n);
 
     if (!lite) {
+        if ((numMeshNodes >= MAX_NUM_NODES) || (memGet.getFreeHeap() < meshtastic_NodeInfoLite_size * 3))
+            cleanupMeshDB();
```

Comment:
I think this should go into a separate PR. Increasing MAX_NUM_NODES or making it configurable is unrelated to the way we handle purging old entries. 
Also, this function should be tested sufficiently to guarantee correct and desired behavior (e.g. sending two successive msgs from new nodes while max_num is reached), so that e.g. ACKs are always sent. Has such a test been done (e.g. with MAX_NUM_NODES set to a very low number like 1 or 2 and check what happens)?

Reply:
Sending ACKs doesn’t require a node to be in the NodeDB. The destination (`to`) will be set according to the `from` field of the received packet.

Reply:
Seems reasonable to pull this, and make it a dedicated PR.